### PR TITLE
Fix #245: Better boolean flag help output

### LINKED ISSUE #245: Better defaults for boolean flags
## 🐛 Bug Report

Boolean flags such as `--compact / --no-compact` with `Default: False` is very confusing.

## To Reproduce

`tartufo -h`

## Expected Behavior

Should show `Default: --no-compact` for `--compact / --no-compact` and similar for other boolean flags

### PR DESCRIPTION
**Pull request by tarkatronic**
To help us get this pull request reviewed and merged quickly, please be sure to include the following items:

* [X] Tests (if applicable)
* [ ] Documentation (if applicable)
* [X] Changelog entry
* [X] A full explanation here in the PR description of the work done

## PR Type
What kind of change does this PR introduce?

* [ ] Bugfix
* [ ] Feature
* [ ] Code style update (formatting, local variables)
* [ ] Refactoring (no functional changes, no api changes)
* [ ] Build related changes
* [ ] CI related changes
* [X] Documentation content changes
* [ ] Tests
* [X] Other

## Backward Compatibility

Is this change backward compatible with the most recently released version? Does it introduce changes which might change the user experience in any way? Does it alter the API in any way?

* [X] Yes (backward compatible)
* [ ] No (breaking changes)

## Issue Linking

Fixes #245 

## What's new?

Using the latest version of `click` gives us better output for our command line boolean flags. For example:
```md
  --entropy / --no-entropy        Enable entropy checks.  [default: True]
```
becomes:
```md
  --entropy / --no-entropy        Enable entropy checks.  [default: entropy]
```

Turns out, there's also a whole ton of other bugfixes and improvements. One of those bugfixes related to path resolution in Windows and resulted in us needing to update some tests. Overall, should be a major positive!

### CODE REVIEW

File: tests/test_cli.py [hunk 1]
```diff
@@ -103,9 +103,11 @@ def test_output_dir_is_called_out(
             result = runner.invoke(
                 cli.main, ["--output-dir", "./foo", "scan-local-repo", "."]
             )
+        # Perform a little bit of trickery to make absolutely certain we get a full absolute path
```
Comment (sushantmimani) [inline tests/test_cli.py]: May be a good idea to link the source of inspiration for this bit of trickery?

**Review by sushantmimani — DISMISSED** [review]
LGTM 🦅

**Review by rbailey-godaddy — APPROVED** [review]
LGTM

**Review by sushantmimani — APPROVED** [review]
LGTM 🦅

**Review by rbailey-godaddy — APPROVED** [review]
even better :)

**Pull request merged**